src/blender_mcp/server.py

```
@@ -524,6 +524,49 @@ def _process_bbox(original_bbox: list[float] | list[int] | None) -> list[int] |
         raise ValueError("Incorrect number range: bbox must be bigger than zero!")
     return [int(float(i) / max(original_bbox) * 100) for i in original_bbox] if original_bbox else None
 
+@mcp.tool("generate_beaver3d_from_text_or_image")
+def generate_beaver3d_from_text_or_image(
+    ctx: Context,
+    text_prompt: str = None,
+    image_url: str = None,
+    position: List[float] = [0, 0, 50],
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)